app/controllers/character_types_controller.rb

```diff
@@ -10,58 +10,54 @@ def index
 
   def new
     @character_type = get_my_character_types.new
-    @character_type.character_attributes.new
   end
 
   def create
     @character_type = new_character_type
 
     if @character_type.save
-      render_show
+      redirect_to @character_type, notice: 'Character type was successfully created.'
     else
-      render :new
+      render :validationerrors
     end
   end
 
   def update
     if @character_type.update(character_type_params)
-      render_show(:update)
+      redirect_to @character_type, notice: 'Character type was successfully updated.'
     else
-      render :edit
+      render :validationerrors
     end
   end
 
   def destroy
     @character_type.destroy
 
     respond_to do |format|
-      format.html { redirect_to character_types_url, notice: 'Character type was successfully destroyed.'}
+      flash.now[:notice] = 'Character type was successfully destroyed.'
+      format.html { redirect_to character_types_url }
       format.js { @character_types = get_my_character_types }
     end
   end
 
   private
 
-    def render_show(method = :create)
-        flash[:notice] = "Character type was successfully #{method}d."
-        render js: "window.location = '#{character_type_path(@character_type)}'"
-    end
-
-    def set_character_type
-      @character_type = current_user.character_types.find_by(id: params[:id])
-      redirect_to(character_types_url, notice: 'Character type was not found.') if @character_type.nil?
-    end
+  def set_character_type
+    @character_type = current_user.character_types.find_by(id: params[:id])
+    redirect_to(character_types_url, notice: 'Character type was not found.') if @character_type.nil?
+    @character_type_id = @character_type.id
```

Comment:
Do you need `@character_type_id` as a separate variable, it seems to me that `@character_type.id` can be used instead?

Reply:
@character_type_id is used in destroy action to remove deleted html element, so @character_type.id can be used but instead of removing element i need to render all types...